src/src/PluginStructs/P036_data_struct.cpp

```diff
@@ -338,32 +339,41 @@ bool P036_data_struct::plugin_write(struct EventStruct *event, const String& str
   int command_i = GetCommandCode(subcommand.c_str(), p036_subcommands);
 
   if (command_i == -1) {
+    String parseString = parseStringKeepCaseNoTrim(string, 3);
+    if (!Parsing) {
+      LineNo = -LineNo;
+      parseString.replace("$$", "%"); // Allow system vars to be passed in by using $$ instead of %
+      parseString.replace("${", "["); // Allow task values to be passed in by using ${ instead of [
+      parseString.replace("}$", "]"); // Allow task values to be passed in by using }$ instead of ]
+    }
```

Comment:
Maybe we should put this in a function with a `byref` parameter, and also apply it to the `userdef1` and `userdef2` commands, from line 588, to make it all work the same (and update documentation, of course).

Reply:
This would be fine, but I was afraid that a change for the header content in `userdef1` and `userdef2 `can effect running applications.
In the header just one **$** is used
`userDef1.replace('$', '%'); // Allow system vars to be passed in by using $ instead of %`

Reply:
That could be a 4th replacement, as a fall-back, and a friendly suggestion in documentation to now use the new `$$`, `${` and `$}` token delimiters, as the old `$` will be deprecated (in the near future).

Reply:
Last commit:
I created a function ReplaceTokenDelimiters() to be used for user defined headers and line content.
The documentation is also changed.
@tonhuisman Is there a way to see the formatted output of *.repl and *.rst units in VS code?

Reply:
> Is there a way or extension to see the formatted output of *.repl and *.rst units in VS code?

The Developers guide has a chapter on [writing documentation](https://espeasy.readthedocs.io/en/latest/Participate/PlatformIO.html#writing-documentation) that also explains how the output can be reviewed in a local browser by opening the `docs/build/html/index.html` file. After a documentation update just run the batchfile (or make html command) again from the docs folder and refresh the browser page.
That works best for me, as I don't have a 4k monitor to effectively use the preview feature of the reStructuredText plugin of VSCode. (Which implies that that is also a usable workflow 😄)

Reply:
Thanks for this info. I've never read it before.
It really helps and works for me, too.

Reply:
Just one question....
Why????
Why adding a new formatting syntax?

I have spent quite a lot of time to get rid of the many copy/paste implementations in the past where each was slightly different and thus causing unpredictable results based on where it is used.

This is in a single plugin, meaning the $ is now only used in this plugin, where in every other place in the code there are several notations to give access to variables.

Reply:
This command not only updates the display contents, but also the internal buffer that is used to update the display after some time. Originally, the interpreted values where stored in the buffer; with this syntax the _variables_ can be stored in the buffer, to be interpreted each time the display is updated. (The part where the text is stored in the buffer is a bit hard to recognize, took me a while to find... 🤔)

Reply:
OK, so we need it as an escape-character to be able to send verbatim internal variables via a command?

If so, then I can see its use, but still this then should not be part of just this plugin as it can be useful for other plugins too.
Or even to be used in commands like `sendTo`, which would trigger a command to be executed on another node via p2p.

Reply:
I think an escape chafacter should be a backslash (thus `\%` for example)
The $ can then later be used for other stuff (e.g. string notations/labels or something like that)

Reply:
That means the suggest notations would be:

- `\%<sysvar>\%` instead of `%<sysvar>%` for system variables
- `\[<TaskName>#<TaskValue>\]` instead of `[<TaskName>#<TaskValue>]` for task values

Reply:
Yep, I did discuss it with Ton this morning.

So in short, this notation using escape-characters then requires a small change in the current parser and adding an extra function to strip the escape characters (e.g. when storing the values)

Reply:
I tried the three escape char sentences.
`\%` works fine, but `\[` and `\]` did not work at all:
1. in Tools->Command I submitted `oledframedcmd,-1,'T: \[myTimer#Value\]s   '`
the logged result was
```
00:15:36.153 : (4640) Info   : HTTP: oledframedcmd,-1,'T: \[myTimer#Value\]s   '
00:15:36.164 : (4472) Info   : Enter P036_data_struct::plugin_write: oledframedcmd,-1,'T: \s   ' (27)
```
where the number in brackets is string.length()

2. in Tools->Command I submitted `oledframedcmd,-1,'T: \\[myTimer#Value\\]s   '`
the logged result was
```
00:17:49.704 : (4760) Info   : HTTP: oledframedcmd,-1,'T: \\[myTimer#Value\\]s   '
00:17:49.714 : (4592) Info   : Enter P036_data_struct::plugin_write: oledframedcmd,-1,'T: \\s   ' (28)
```
Which function is parsing inbetween submitting the command and P036_data_struct::plugin_write?

Reply:
Those are not yet implemented....
The problem I would like to avoid is that we won't end up with separate formatting rules.

The replacement of system variables etc. are done in the `String parseTemplate(String& tmpString)` function.

Reply:
So that would end up in a couple of tasks:
- Don't process variables wrapped in escape sequences when processing template variables (replace by placeholders before processing, restore after processing)
- Provide a `fixupEscapedVariables()` (better name to be proposed) function that removes the escape characters
- Call that function from plugins that need the escaped variables turned into normal variables for further processing (mostly display related, I think, though there may be other uses)

Reply:
`stripEscapeCharacters` would probably be a better name for the function.

Reply:
Just now I comitted a version with the new functions `stripEscapeCharacters()` and `hasEscapeCharacters()`

Reply:
What happens when you try to send some string with those escape characters to log?
For example:
```
logentry,"Systime: \%systime\% without escapechar: %systime%"
```